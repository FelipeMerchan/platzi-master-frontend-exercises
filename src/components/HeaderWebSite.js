import React from 'react'

import MenuWebSite from './MenuWebSite'
import './styles/HeaderWebSite.css'

function HeaderWebSite() {
  return (
    <header className="HeaderWebSite text-left">
        <h1 className="HeaderWebSite__title">Ejercicio Web</h1>
        <h4 className="HeaderWebSite__subtitle">PÃ¡gina creada con HTML5</h4>
        <MenuWebSite />
    </header>
  )
}

export default HeaderWebSite