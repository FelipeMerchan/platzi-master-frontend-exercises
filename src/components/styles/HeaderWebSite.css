.HeaderWebSite {
  padding-top: var(--space-2);
  font-family: var(--font-2);
  color: var(--typographyColor);
}

.HeaderWebSite__title {
  font-size: 32px;
  margin: 0 0 var(--space);
}

.HeaderWebSite__subtitle {
  margin: 0 0 var(--space);
}