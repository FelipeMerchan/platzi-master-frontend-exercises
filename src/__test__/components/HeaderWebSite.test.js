import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import HeaderWebSite from '../../components/HeaderWebSite'

describe('Probar el componente <HeaderWebSite />', () => {
  const headerWebSite = shallow(<HeaderWebSite />)

  test('Render del componente HeaderWebSite', () => {
    expect(headerWebSite.length).toEqual(1)
  })

  test('Render de los 3 hijos', () => {
    expect(headerWebSite.children()).toHaveLength(3);
  });

  test('Render del tÃ­tulo', () => {
    expect(headerWebSite.find('.HeaderWebSite__title').text()).toEqual('Ejercicio Web')
  })
})

describe('HeaderWebSite Snapshot', () => {
  test('Comprobar la interfaz de usuario del componente HeaderWebSite', () => {
    const headerWebSite = create(<HeaderWebSite />)
    expect(headerWebSite.toJSON()).toMatchSnapshot()
  })
})