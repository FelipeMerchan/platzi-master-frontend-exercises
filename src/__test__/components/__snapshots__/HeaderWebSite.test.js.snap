// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderWebSite Snapshot Comprobar la interfaz de usuario del componente HeaderWebSite 1`] = `
<header
  className="HeaderWebSite text-left"
>
  <h1
    className="HeaderWebSite__title"
  >
    Ejercicio Web
  </h1>
  <h4
    className="HeaderWebSite__subtitle"
  >
    PÃ¡gina creada con HTML5
  </h4>
  <nav
    className="MenuWebSite"
  >
    <ul>
      <li
        className="MenuWebSite_item"
      >
        <a
          className="is-active"
          href="#home"
          id="home"
          onClick={[Function]}
          role="button"
        >
          Home
        </a>
      </li>
      <li
        className="MenuWebSite_item"
      >
        <a
          href="#aboutUs"
          id="aboutUs"
          onClick={[Function]}
          role="button"
        >
          Acerca de
        </a>
      </li>
      <li
        className="MenuWebSite_item"
      >
        <a
          href="#portfolio"
          id="portfolio"
          onClick={[Function]}
          role="button"
        >
          Portafolio
        </a>
      </li>
      <li
        className="MenuWebSite_item"
      >
        <a
          href="#blog"
          id="blog"
          onClick={[Function]}
          role="button"
        >
          Blog
        </a>
      </li>
      <li
        className="MenuWebSite_item"
      >
        <a
          href="#contactUs"
          id="contactUs"
          onClick={[Function]}
          role="button"
        >
          Contacto
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
